cpp/src/arrow/dataset/file_base.cc

```diff
@@ -71,12 +70,16 @@ Status FileSystemBasedDataSource::Make(fs::FileSystem* filesystem,
   stats.resize(new_end - stats.begin());
 
   out->reset(new FileSystemBasedDataSource(filesystem, selector, std::move(format),
-                                           std::move(scan_options), std::move(stats)));
+                                           std::move(stats)));
   return Status::OK();
 }
 
 std::unique_ptr<DataFragmentIterator> FileSystemBasedDataSource::GetFragments(
     std::shared_ptr<ScanOptions> options) {
+  if (auto empty = AssumeCondition(&options)) {
```

Comment:
isn't that C++14?

Reply:
No; so far as I know one can use a declaration as an if statement condition all the way back to c++03